wiki/Introduction-to-Creating-a-Job.md

```diff
@@ -0,0 +1,1818 @@
+
+Note that this walkthrough is written for a bash-compatible shell such as those available on typical Linux or
+MacOS workstations. If you are using Windows, then we trust that the modifications to powershell or batch from bash
+will be straightforward enough for you to handle; though you may also run bash scripts on Windows if you install
+an application such as [Git BASH](https://gitforwindows.org/).
+
+#### 2.1.1. Creating a shell script to render with Blender
+
+First, you'll create a shell script that runs Blender to render a part of the animation.
+Looking up the [documentation for the Blender command line](https://docs.blender.org/manual/en/latest/advanced/command_line/arguments.html)
+we created the following as a starting point for our Blender script:
+
+```bash
+#!/bin/bash
+
+# Return an error code if any command in the script fails.
+set -eou pipefail
+
+# Use Blender's scripting interface to reduce the scene resolution and sampling rate to speed up testing.
+# Remove this after testing is complete.
+TMPFILE=$(mktemp)
+cat > "$TMPFILE" << EOF
+import bpy
+for s in bpy.data.scenes:
+  s.render.resolution_x = 480
+  s.render.resolution_y = 270
+bpy.context.scene.cycles.samples = 100
+EOF
+
+# Run blender to generate the image frames.
+blender --background 3d/pavillon_barcelone_v1.2.blend \
+  --python "$TMPFILE" \
+  --render-output output_frames/frame-### \
+  --render-format PNG --use-extension 1 \
+  --render-frame "1..2"
+
+# Cleanup the temporary python script
+rm -f $TMPFILE
+```
+
+This script makes some assumptions, such as being run from a directory that contains our input and output, that we'll modify next, but
+this is a good start. It uses Blender to render two frames of the 380 frame animation created by
+the ["Barcelona Pavillion" demo scene](https://www.blender.org/download/demo-files/#cycles) (accessed July 2024) at reduced resolution.
+Save this to a file called `render.sh`, set its execute bit (`chmod +x render.sh`), and then test it to verify that it runs as expected:
+
+```bash
+% ./render.sh
+Blender 4.1.1 (hash e1743a0317bc built 2024-04-16 00:06:22)
+...
+Fra:1 Mem:130.47M (Peak 206.18M) | Time:00:07.56 | Compositing | De-initializing execution
+Saved: 'output_frames/frame-001.png'
+...
+Fra:2 Mem:130.48M (Peak 272.58M) | Time:00:07.64 | Compositing | De-initializing execution
+Saved: 'output_frames/frame-002.png'
+Time: 00:07.71 (Saving: 00:00.07)
+```
+
+Next, you'll want to be able to create a Job that can render any scene rather than just this one demo scene. So, parameterize the script
+such that the scene file, output directory, and frames to render can all be passed in as arguments to the script:
+
+```bash
+#!/bin/bash
+
+# Return an error code if any command in the script fails.
+set -eou pipefail
+
+TMPFILE=$(mktemp)
+cat > "$TMPFILE" << EOF
+import bpy
+for s in bpy.data.scenes:
+  s.render.resolution_x = 480
+  s.render.resolution_y = 270
+bpy.context.scene.cycles.samples = 100
+EOF
+
+SCENE="$1"
```

Comment:
Recommend added a comment line above this which quickly explains that `$1` etc will translate into arguments on the command line calling the script, with the script itself being `$0`